_posts/2019-02-08-resize-images-on-the-fly.md

```diff
@@ -0,0 +1,187 @@
+
+(Image here)
+
+The default sample function serving traffic in the OpenFaaS Portal.
+
+2. Write the resizing logic using sharp
+
+We are going to use the sharp package to resize the images. This is where we would have normally used ImageMagick or GraphicsMagick. This is what the sharp website says about the library:
+
+> The typical use case for this high speed Node.js module is to convert large images in common formats to smaller, web-friendly JPEG, PNG and WebP images of varying dimensions.
+> Resizing an image is typically 4x-5x faster than using the quickest ImageMagick and GraphicsMagick settings.
+
+So at IconScout, having a 4x-5x speed increase was a great move forward for us where we process 10s of thousands of images per day and that’s growing every week.
+
+Now let’s install the sharp package using `npm`.
+
+```bash
+$ cd ~/dev/resizer/image-resizer
+npm install --save sharp
+```
+We’ll also need the request package to fetch the specified image from its URL.
+
+```bash
+$ cd ~/dev/resizer/image-resizer
+$ npm install --save request
+```
+
+Resizing images with sharp is really easy and simple to use, here is a link to the documentation: http://sharp.dimens.io/en/stable/api-resize/
+
+For example, to resize input.png to 200px x 200px, you only need the following code:
+
+```js
+sharp('input.png')
+  .png()
+  .resize(200, 200, {
+    fit: 'contain'
+  })
+  .toFile('output.png')
+```
+
+Let’s wrap this resizing code to our function.
+
+You can find the full code in this GitHub repo: https://github.com/tarunmangukiya/openfaas-functions/blob/master/image-resize/handler.js
+
+
+The way our function will work, is that the user will supply all the parameters on a query-string, and that means we can even embed resized images pointing directly at the function.
+
+An example URL might be: http://127.0.0.1:8080/function/image-resize?height=300&width=300&url=https://upload.wikimedia.org/wikipedia/commons/3/36/Hopetoun_falls.jpg
+
+So we will be fetching the image from the URL and then resizing it to the provided dimensions.
+
+These are all the parameters I’ve added:
+
+*url:* Public URL of image
+*width:* Number
+*height:* Number
+*download:* Download image rather than showing in browser
+*fit:* Fit the image in the given width & height
+
+See all the other options available http://sharp.dimens.io/en/stable/api-resize/
+
+Update your function with the code from my repo, then deploy the function with the faas-cli.
+
+```bash
+$ cd ~/dev/resizer/
+$ faas-cli up -f image-resize.yml
+```
+
+Let’s test it all out
+
+I’ve found a picture of a waterfall on Wikipedia and I want to resize it to 300px x 300px.
+
+https://upload.wikimedia.org/wikipedia/commons/3/36/Hopetoun_falls.jpg
+
+You could also find your own favourite images to try out.
+
+(IMAGE)
+
+Try it out live with your own OpenFaaS where `$OPENFAAS_URL` is the IP address and port of your gateway. On Swarm this is usually `http://127.0.0.1:8080`
```

Comment:
You could use `faas-cli invoke` with the `--query` flag to do this.


```sh
echo ""  | faas-cli -f image-resize.yml invoke image-resize  --query height=300 --query width=300 --query url=https://upload.wikimedia.org/wikipedia/commons/3/36/Hopetoun_falls.jpg
```

Reply:
Yees this can be more useful. @alexellis